### ISSUE TITLE
How can we control the order of rendering of paths in swagger-ui?

### ISSUE THREAD

**Issue by mandeepm91** [kogosoftwarellc/open-api#335]
I have my paths directory structure as follows:

```
├── cities.js
├── auth
│   └── login.js
├── users
    └── profile.js

```

When this gets rendered in swagger-ui, the paths can appear in some random sequence. I want to somehow make sure that all the paths related to auth are displayed first, then users paths and then cities paths. Is there any way I can configure this in express-openapi ?

**Comment by jsdevel** [comment]
See byRoute in https://github.com/kogosoftwarellc/open-api/blob/master/packages/openapi-framework/src/util.ts.

**Comment by mandeepm91** [comment]
Thanks. So if I comment out this line https://github.com/kogosoftwarellc/open-api/blob/b01cc368c94cff4646d328dabf5f55a9b8578d6a/packages/openapi-framework/index.ts#L229

and then provide the `paths` as an array of objects with `path` and `module`, the ordering that I provide is preserved. What are your thoughts on the idea of making the sorting configurable in the `initialize` function? Like if we provide an argument like this:

```
initialize({
  app,
  apiDoc: v1ApiDoc,
  paths: './api-v1/paths',

  pathSortingFn: function (a, b) {
    // some sorting function 
    // defaults to byRoute
  }

});
```

**Comment by jsdevel** [comment]
@mandeepm91 i like it!  feel free to open a PR!

**Comment by mandeepm91** [comment]
I've made my changes and ready to commit. I was going through the workflow and it says not to bump the version in `package.json` file. However, when I run the tests, it does alter the `package-lock.json` file. Should I keep those while committing or discard them?

**Comment by jsdevel** [comment]
I'll bump the version.  Just be sure to add tests.

**Comment by jsdevel** [comment]
Updates to package-lock.json are fine.

**Comment by mandeepm91** [comment]
Did some more research on this and it seems this won't be a good idea. The rendered JSON has paths as an object and not an array so even though my fix works, theoretically it can break since ordering of keys in a Javascript object is not guaranteed. Also, I found that we can use `tags` to group the APIs together and configure their sorting in swagger-ui. Although, in my opinion it would have been nicer if the ordering in the tags array dictated the sorting of the tags (as mentioned in the OAS 3 documentation) but that doesn't seem to be the case. I tried both swagger-ui and ReDoc and both of them sort tags alphabetically by default while rendering, irrespective of the order in which they are provided in the tags array in the spec. But the good thing is that it can be configured by passing an `apiSorter` function in swagger-ui config. I'll try that. I am closing this issue since the purpose of this package is to generate the spec and spec does not dictate the ordering or APIs.
